### PR TITLE
Migrate group show to Bootstrap

### PR DESCRIPTION
**Pull request by eduardoj**
Updating the group members is not done yet. It will be done in another pull request.

2018-11-29-group_show

### CODE REVIEW

File: src/api/app/views/webui2/webui/groups/show.html.haml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      = render(partial: 'webui2/shared/requests_table', locals: { id: 'all_requests_table', source_url: group_requests_path(@group) })
+
+    .tab-pane#group-members{ aria: { controls: 'group-members' }, role: 'tabpanel' }
+      %h3 Group Members
+      %table.responsive.table.table-striped.table-bordered#group-members-table{ width: '100%' }
+        %thead
+          %tr
+            %th Login Name
+        %tbody
+          - if @group.users.any?
```
[the remark this answers is not in the document]
Comment (bgeuken) [reply src/api/app/views/webui2/webui/groups/show.html.haml]: To see that there is no content? :thinking:

I think I would find it confusing if the table some times is there and sometimes it's not.  Since that's the main job of this tab, showing the group members.

File: src/api/app/views/webui2/webui/groups/show.html.haml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      %h3 Incoming Requests
+      = render(partial: 'webui2/shared/requests_table', locals: { id: 'requests_in_table', source_url: group_requests_path(@group) })
+
+    .tab-pane#all-requests{ aria: { controls: 'all-requests' }, role: 'tabpanel' }
+      %h3 All Requests
+      = render(partial: 'webui2/shared/requests_table', locals: { id: 'all_requests_table', source_url: group_requests_path(@group) })
+
+    .tab-pane#group-members{ aria: { controls: 'group-members' }, role: 'tabpanel' }
+      %h3 Group Members
+      %table.responsive.table.table-striped.table-bordered#group-members-table{ width: '100%' }
```
Comment (dmarcoux) [inline src/api/app/views/webui2/webui/groups/show.html.haml]: `width: 100%` can be replaced by the Bootstrap class `w-100`

File: src/api/app/views/webui2/shared/_requests_table.html.haml [hunk 3]
```diff
@@ -10,4 +10,4 @@
       %th Type
       %th Priority
       %th
-%tbody
+  %tbody
```
[the remark this answers is not in the document]
Comment (eduardoj) [reply src/api/app/views/webui2/shared/_requests_table.html.haml]: > this is still wrong indented confused

Hm... I don't think so. Can you click on "View file" and check again?

File: src/api/app/views/webui2/webui/groups/show.html.haml (hunk 2 above)
Comment (eduardoj) [reply src/api/app/views/webui2/webui/groups/show.html.haml]: > `width: 100%` can be replaced by the Bootstrap class `w-100`

Done.

File: src/api/app/views/webui2/webui/groups/show.html.haml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+              %td This group does not contain users.
+      - if User.current.is_admin? || @group.group_maintainers.where(user: User.current).exists?
+        = link_to group_edit_title_path(title: @group) do
+          %i.fas.fa-edit
+          Edit group members
+
+= content_for :ready_function do
+  :plain
+    $('#group-members-table').DataTable({
+      responsive: true
```
[the remark this answers is not in the document]
Comment (eduardoj) [reply src/api/app/views/webui2/webui/groups/show.html.haml]: > why is the responsive necessary? :confused:

I used what stays in the [table page](https://obs-patterns.netlify.com/atoms/table.html) of our patterns, for a Table with DataTable.

**Comment by dmarcoux** [comment]
To add to what @bgeuken, we also have the tabs in the [patterns](https://obs-patterns.netlify.com/atoms/tabs.html). Once you add the dropdown, it should be working as you already have the `collapsible` class on the `ul` tag.

File: src/api/app/views/webui2/webui/groups/show.html.haml (hunk 1 above)
[the remark this answers is not in the document]
Comment (eduardoj) [reply src/api/app/views/webui2/webui/groups/show.html.haml]: This is how it will look like with the current code for a group without users:

group_members_empty

I'm with @bgeuken.

File: src/api/app/views/webui2/webui/groups/show.html.haml [hunk 5]
```diff
@@ -0,0 +1,56 @@
+- @pagetitle = "Group #{@group.title}"
```
[the remark this answers is not in the document]
Comment (eduardoj) [reply src/api/app/views/webui2/webui/groups/show.html.haml]: > `.title` is not needed

Removed.

File: src/api/app/views/webui2/webui/groups/show.html.haml [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+      %h3 Group Members
+      %table.responsive.table.table-striped.table-bordered#group-members-table{ width: '100%' }
+        %thead
+          %tr
+            %th Login Name
+        %tbody
+          - if @group.users.any?
+            - @group.users.each do |user|
+              %tr
+                %td= link_to(user.login, user_show_path(user))
```
[the remark this answers is not in the document]
Comment (eduardoj) [reply src/api/app/views/webui2/webui/groups/show.html.haml]: > `.login` should be needed

If you wanted to say `should not be needed`, yes. Removed.

File: src/api/app/views/webui2/webui/groups/show.html.haml (hunk 1 above)
[the remark this answers is not in the document]
Comment (dmarcoux) [reply src/api/app/views/webui2/webui/groups/show.html.haml]: I agree with @bgeuken and @eduardoj

**Comment by eduardoj** [comment]
@Ana06:

> from the first commit:
> 
> > from configuration to Bootstrap
> 
> what does this mean? :confused:

I removed it `from configuration` in the description of the pull request. I realized that could be confusing there, but forgot to remove it in the commit message. Removed from the commit message too.

> Was the second commit breaking something? I would say so. Can you include this information in the commit message?

It was breaking the indentation. I will add some more comment to clarify the what is th outputs in both cases.

> Tests need to be addapted

Sure. I'm working on it.

**Comment by eduardoj** [comment]
@bgeuken:

> The tabs disappear on smaller screens. We had the same issue. You gone need a dropdown

Done. It was esasy! :+1:

File: src/api/app/views/webui2/webui/groups/show.html.haml (hunk 4 above)
[the remark this answers is not in the document]
Comment (Ana06) [reply src/api/app/views/webui2/webui/groups/show.html.haml]: And I asked why it is needed (not where you copied from) :rofl: either you do it with the responsible class or with the parameter, but not both. It is an unnecessary redundancy (this should also be updated in the patterns, I'll take care) :unamused:

File: src/api/app/views/webui2/webui/groups/show.html.haml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ana06) [reply src/api/app/views/webui2/webui/groups/show.html.haml]: We do it differently in other pages such as the user home page page. We need some application consistency here.

File: src/api/app/views/webui2/shared/_requests_table.html.haml (hunk 3 above)
[the remark this answers is not in the document]
Comment (Ana06) [reply src/api/app/views/webui2/shared/_requests_table.html.haml]: Github bug! :bug:

File: src/api/app/views/webui2/webui/groups/show.html.haml (hunk 4 above)
[the remark this answers is not in the document]
Comment (Ana06) [reply src/api/app/views/webui2/webui/groups/show.html.haml]: https://datatables.net/reference/option/responsive#Default

[the remark this answers is not in the document]
Comment (Ana06) [reply src/api/app/views/webui2/webui/groups/show.html.haml]: https://datatables.net/extensions/responsive/init

[the remark this answers is not in the document]
Comment (eduardoj) [reply src/api/app/views/webui2/webui/groups/show.html.haml]: Parameter removed. :+1:

**Comment by eduardoj** [comment]
> > Tests need to be addapted
> 
> Sure. I'm working on it.

Done. I think I've addressed all the changes requested. @Ana06, @bgeuken, @dmarcoux please have another look.

File: src/api/app/views/webui2/webui/groups/show.html.haml [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+          - else
+            %tr
+              %td This group does not contain users.
+      - if User.current.is_admin? || @group.group_maintainers.where(user: User.current).exists?
+        = link_to group_edit_title_path(title: @group) do
+          %i.fas.fa-edit
+          Edit group members
+
+= content_for :ready_function do
+  $('#group-members-table').DataTable();
```
Comment (Ana06) [inline src/api/app/views/webui2/webui/groups/show.html.haml]: I think may PR (https://github.com/openSUSE/open-build-service/pull/6410) will get merged before than yours, so you may want to use the new pattern: https://github.com/openSUSE/obs-patterns/pull/42

Comment (eduardoj) [reply src/api/app/views/webui2/webui/groups/show.html.haml]: I suppose that otherwise you will not approve the PR. Let's wait then.

Comment (eduardoj) [reply src/api/app/views/webui2/webui/groups/show.html.haml]: #6410 was merged. New pattern used.

File: src/api/app/views/webui2/webui/groups/show.html.haml [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+      %h3 Incoming Requests
+      = render(partial: 'webui2/shared/requests_table', locals: { id: 'requests_in_table', source_url: group_requests_path(@group) })
+
+    .tab-pane#all-requests{ aria: { controls: 'all-requests' }, role: 'tabpanel' }
+      %h3 All Requests
+      = render(partial: 'webui2/shared/requests_table', locals: { id: 'all_requests_table', source_url: group_requests_path(@group) })
+
+    .tab-pane#group-members{ aria: { controls: 'group-members' }, role: 'tabpanel' }
+      %h3 Group Members
+      %table.responsive.table.table-striped.table-bordered.w-100#group-members-table
```
Comment (Ana06) [inline src/api/app/views/webui2/webui/groups/show.html.haml]: Drop repsosive, it is not working properly

File: src/api/app/views/webui2/webui/groups/show.html.haml [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+          - else
+            %tr
+              %td This group does not contain users.
+      - if User.current.is_admin? || @group.group_maintainers.where(user: User.current).exists?
+        = link_to group_edit_title_path(title: @group) do
+          %i.fas.fa-edit
+          Edit group members
+
+= content_for :ready_function do
+  initializeDataTable('#group-members-table');
```
Comment (Ana06) [inline src/api/app/views/webui2/webui/groups/show.html.haml]: you need `obs-dataTable` otherwise the fix doesn't work. Please check: https://github.com/openSUSE/obs-patterns/pull/42

File: src/api/app/views/webui2/webui/groups/show.html.haml (hunk 8 above)
Comment (eduardoj) [reply src/api/app/views/webui2/webui/groups/show.html.haml]: Done.

File: src/api/app/views/webui2/webui/groups/show.html.haml (hunk 9 above)
Comment (eduardoj) [reply src/api/app/views/webui2/webui/groups/show.html.haml]: Done.

Comment (Ana06) [reply src/api/app/views/webui2/webui/groups/show.html.haml]: The class is not in the correct place. Have you tried it? :confused:

File: src/api/app/views/webui2/webui/groups/show.html.haml [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+          %tbody
+            - if @group.users.any?
+              - @group.users.each do |user|
+                %tr
+                  %td= link_to(user, user_show_path(user))
+            - else
+              %tr
+                %td This group does not contain users.
+        - if User.current.is_admin? || @group.group_maintainers.where(user: User.current).exists?
+          = link_to group_edit_title_path(title: @group) do
```
Comment (dmarcoux) [inline src/api/app/views/webui2/webui/groups/show.html.haml]: This should also be handled to complete the card, but it can be in another PR.

Comment (dmarcoux) [reply src/api/app/views/webui2/webui/groups/show.html.haml]: I created a card so we don't forget it: https://trello.com/c/MW1IGcM7/987-edit-group-members
